console/console-init/ui/src/modules/iot-device/dialogs/CreateDevice/AddDeviceUsingJsonPage.tsx

```diff
@@ -39,8 +44,19 @@ export default function AddDeviceUsingJsonPage() {
 
   useBreadcrumb(breadcrumb);
 
-  const onSave = (detail: string) => {
+  const onSave = async (_detail: string) => {
     //TODO: Add query to save iot device
+    if (deviceDetail) {
+      const device: ICreateDeviceRequest = getDeviceFromDeviceString(
+        deviceDetail
+      );
+      const variable = {
+        iotproject: { name: projectname, namespace },
+        device: device
+      };
+      console.log(variable);
+      await setCreateDeviceQueryVariables(variable);
+    }
     history.push(deviceListRouteLink);
```

Comment:
I think you should call change route code in useMutation query as a callback function based on success or failure.